actionpack/lib/abstract_controller/rendering.rb

```diff
@@ -64,12 +64,12 @@ def rendered_format
     # You can overwrite this configuration per controller.
     def view_assigns
       protected_vars = _protected_ivars
-      variables      = instance_variables
 
-      variables.reject! { |s| protected_vars.include? s }
-      variables.each_with_object({}) { |name, hash|
-        hash[name.slice(1, name.length)] = instance_variable_get(name)
-      }
+      instance_variables.each_with_object({}) do |name, hash|
+        unless protected_vars.include?(name)
+          hash[name.slice(1, name.length)] = instance_variable_get(name)
```

Comment:
`delete_prefix("@")` is another alternative (clearer too).

Reply:
I agree that it's clearer, but we lose some of the gains because that method only exists for String and `name` here is a Symbol.